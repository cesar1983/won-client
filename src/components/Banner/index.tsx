import Button from 'components/Button'
import Ribbon from 'components/Ribbon'
import * as S from './styles'

export type BannerProps = {
  img: string
  title: string
  subtitle: string
  buttonLabel: string
  buttonLink: string
  ribbon?: string
  ribbonColor: 'primary' | 'secondary'
  ribbonSize: 'normal' | 'small'
}

const Banner = ({
  img,
  title,
  subtitle,
  buttonLabel,
  buttonLink,
  ribbon,
  ribbonColor,
  ribbonSize
}: BannerProps) => (
  <S.Wrapper>
    <S.Image src={img} role="img" aria-label={title} />
    {!!ribbon && (
      <Ribbon color={ribbonColor} size={ribbonSize}>
        {ribbon}
      </Ribbon>
    )}
    <S.Caption>
      <S.Title>{title}</S.Title>
      <S.Subtitle dangerouslySetInnerHTML={{ __html: subtitle }} />
      <Button as="a" href={buttonLink} size="large">
        {buttonLabel}
      </Button>
    </S.Caption>
  </S.Wrapper>
)

export default Banner
