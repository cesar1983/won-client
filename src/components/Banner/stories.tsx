import { Story, Meta } from '@storybook/react/types-6-0'
import Banner, { BannerProps } from '.'

export default {
  title: 'Banner',
  component: Banner,
  args: {
    img:
      'https://venturebeat.com/wp-content/uploads/2020/01/doom-eternal-4.jpg',
    title: 'Doom',
    subtitle: '<i>Eternal Yehhhh</i>',
    buttonLabel: 'Buy now',
    buttonLink: '/games/doom'
  }
} as Meta

export const Default: Story<BannerProps> = (args) => <Banner {...args} />
