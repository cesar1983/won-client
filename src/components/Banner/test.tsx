import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import Banner from '.'

const props = {
  img: 'https://venturebeat.com/wp-content/uploads/2020/01/doom-eternal-4.jpg',
  title: 'Doom Final',
  subtitle: '<i>Eternal <strong>Yehhhh</strong> Doommmm</i>',
  buttonLabel: 'Buy now',
  buttonLink: '/games/doom'
}

describe('<Banner />', () => {
  it('should render correctly', () => {
    const { container } = renderWithTheme(<Banner {...props} />)

    expect(
      screen.getByRole('heading', { name: /Doom Final/i })
    ).toBeInTheDocument()

    expect(
      screen.getByRole('heading', { name: /Eternal/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('img', { name: /Doom/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
