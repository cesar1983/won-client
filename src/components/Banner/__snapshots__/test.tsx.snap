// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render correctly 1`] = `
.c5 {
  cursor: pointer;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient(180deg,#ff5f5f 0%,#f062c0 50%);
  color: #FAFAFA;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c5:hover {
  background: linear-gradient(180deg,#e35565 0%,#d958a6 50%);
}

.c0 {
  position: relative;
}

.c1 {
  width: 100%;
  height: 23rem;
  background-color: #EAEAEA;
  background-image: url(https://venturebeat.com/wp-content/uploads/2020/01/doom-eternal-4.jpg);
  background-position: center center;
  background-size: cover;
}

.c2 {
  width: 100%;
  background-color: rgba(0,0,0,0.7);
  padding: 2.4rem;
}

.c3 {
  font-size: 1.8rem;
  font-weight: 600;
  color: #FAFAFA;
}

.c4 {
  font-size: 1.4rem;
  font-weight: 400;
  color: #FAFAFA;
  margin-bottom: 1.6rem;
}

.c4 strong {
  font-weight: 600;
  color: #F231A5;
}

@media (min-width:768px) {
  .c0 {
    box-shadow: 0 0.4rem 0.5rem 0 rgba(0,0,0,0.2);
  }
}

@media (min-width:768px) {
  .c1 {
    height: 58rem;
  }
}

@media (min-width:768px) {
  .c2 {
    border-radius: 0 0 0.4rem 0.4rem;
    padding: 4.0rem;
    position: absolute;
    bottom: 0;
    left: 0;
  }
}

@media (min-width:768px) {
  .c3 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c4 {
    font-size: 1.8rem;
  }
}

<main
  class="c0"
>
  <div
    aria-label="Doom Final"
    class="c1"
    role="img"
    src="https://venturebeat.com/wp-content/uploads/2020/01/doom-eternal-4.jpg"
  />
  <div
    class="c2"
  >
    <h2
      class="c3"
    >
      Doom Final
    </h2>
    <h3
      class="c4"
    >
      <i>
        Eternal 
        <strong>
          Yehhhh
        </strong>
         Doommmm
      </i>
    </h3>
    <a
      class="c5"
      href="/games/doom"
    >
      <span>
        Buy now
      </span>
    </a>
  </div>
</main>
`
